src/main/groovy/org/transmartproject/batch/tag/TagTypeFlatFileReader.groovy

```diff
@@ -0,0 +1,60 @@
+package org.transmartproject.batch.tag
+
+import org.springframework.batch.item.ItemStreamReader
+import org.springframework.batch.item.file.ResourceAwareItemReaderItemStream
+import org.springframework.batch.item.file.mapping.FieldSetMapper
+import org.springframework.batch.item.file.transform.DelimitedLineTokenizer
+import org.springframework.batch.item.file.transform.FieldSet
+import org.springframework.core.io.Resource
+import org.springframework.validation.BindException
+import org.transmartproject.batch.beans.StepBuildingConfigurationTrait
+
+import javax.annotation.PostConstruct
+
+/**
+ *
+ */
+class TagTypeTsvFileReader implements ItemStreamReader<TagType>, StepBuildingConfigurationTrait  {
```

Comment:
Don't use `StepBuildingConfigurationTrait`... This is to be used in `@Configuration` classes, as evidenced by all the methods that it contains and that you're copying to this class. Fixing this is simple, create a factory method in a `@Configuration` class and make `TagTypeMapper` a top level (it doesn't have any dependence on the outer class here, and in fact should have been marked `static`), which is something that you seem oddly reluctant to do, judging from all the classes that are mere wrappers around the real `@Delegate`d artifact.